test/e2e/commands/xmlupload/test_values.py

```diff
@@ -0,0 +1,293 @@
+    assert actual_value == expected_val
+    assert next(res_g.objects(val_iri, RDF.type)) == KNORA_API.GeonameValue
+    assert len(val_triples) == BASE_NUMBER_OF_TRIPLES_PER_VALUE
+
+
+def test_integer(res_g, onto_iri):
+    prop_iri = URIRef(f"{onto_iri}testIntegerSimpleText")
+    val_iri = _assert_number_of_values_is_one_and_get_val_iri(res_g, "integer", prop_iri)
+    val_triples = list(res_g.triples((val_iri, None, None)))
+    expected_val = Literal(1)
+    actual_value = next(res_g.objects(val_iri, KNORA_API.intValueAsInt))
+    assert actual_value == expected_val
+    assert next(res_g.objects(val_iri, RDF.type)) == KNORA_API.IntValue
+    assert len(val_triples) == BASE_NUMBER_OF_TRIPLES_PER_VALUE
+
+
+def test_interval(auth_header, project_iri, creds):
+    cls_iri_str = f"{KNORA_API_STR}AudioSegment"
+    res_g = util_request_resources_by_class(cls_iri_str, auth_header, project_iri, creds)
+    val_iri = _assert_number_of_values_is_one_and_get_val_iri(res_g, "audio_segment", KNORA_API.hasSegmentBounds)
+    val_triples = list(res_g.triples((val_iri, None, None)))
+    interval_start = Literal("0.1", datatype=XSD.decimal)
+    actual_interval_start = next(res_g.objects(val_iri, KNORA_API.intervalValueHasStart))
+    assert actual_interval_start == interval_start
+    interval_end = Literal("0.2", datatype=XSD.decimal)
+    actual_interval_end = next(res_g.objects(val_iri, KNORA_API.intervalValueHasEnd))
+    assert actual_interval_end == interval_end
+    assert next(res_g.objects(val_iri, RDF.type)) == KNORA_API.IntervalValue
+    additional_interval_triple = 1
+    assert len(val_triples) == BASE_NUMBER_OF_TRIPLES_PER_VALUE + additional_interval_triple
+
+
+def _util_get_list_node(creds, auth_header) -> str:
+    all_list_one_project_endpoint = f"{creds.server}/admin/lists?9999"
+    headers = auth_header | {"Content-Type": "application/json"}
+    response_all_lists = requests.get(all_list_one_project_endpoint, timeout=3, headers=headers).json()
+    first_list = next(x for x in response_all_lists["lists"] if x["name"] == "firstList")
+    list_iri = urllib.parse.quote_plus(first_list["id"])
+    all_nodes_one_list_endpoint = f"{creds.server}/admin/lists/{list_iri}"
+    response_all_nodes = requests.get(all_nodes_one_list_endpoint, timeout=3, headers=headers).json()
+    children = response_all_nodes["list"]["children"]
+    node_one = next(x for x in children if x["name"] == "n1")
+    return cast(str, node_one["id"])
+
+
+def test_list(res_g, onto_iri, creds, auth_header):
+    prop_iri = URIRef(f"{onto_iri}testListProp")
+    val_iri = _assert_number_of_values_is_one_and_get_val_iri(res_g, "list", prop_iri)
+    val_triples = list(res_g.triples((val_iri, None, None)))
+    list_node_iri = _util_get_list_node(creds, auth_header)
+    expected_val = URIRef(list_node_iri)
+    actual_value = next(res_g.objects(val_iri, KNORA_API.listValueAsListNode))
+    assert actual_value == expected_val
```

Comment:
Here you compare 2 different API responses from different endpoints. That means, you test if one API endpoint gives the same info than the other. I think you should rather test against a hardcoded string, which you copy-paste from the XML file:

```suggestion
    actual_value = next(res_g.objects(val_iri, KNORA_API.listValueAsListNode))
    assert actual_value == "n1"
```

Reply:
I disagree. We are not testing the endpoint. If we want to do that then whe should test it elsewhere. I would like to test if we replaced the IRI correctly. Unfortunately this is only possible by querying ourselves.